pgxn/neon_walredo/walredoproc.c

```diff
@@ -226,7 +243,146 @@ WalRedoMain(int argc, char *argv[])
 	InitializeWalConsistencyChecking();
 #endif
 
-	CreateSharedMemoryAndSemaphores();
+	/*
+	 * Initialize dummy shmem.
+	 *
+	 * This code follows CreateSharedMemoryAndSemaphores() but manually sets up
+	 * the shmem header and skips few initializations that are not needed for
+	 * WAL redo.
+	 */
+	PGShmemHeader *shim = NULL;
+	{
+		PGShmemHeader *hdr;
+		Size size = 10*1048576; // 20 mb. Default nbuffers are 1024, so 8mb shmem
```

Comment:
We could call CalculateShmemSize() for this, like CreateSharedMemoryAndSemaphores does

Reply:
CalculateShmemSize is pg-v15 only, so used it in case of v15 and hardcoded results of the same calculation for v14